client/my-sites/marketing/connections/service.jsx

```diff
@@ -263,20 +266,33 @@ export class SharingService extends Component {
 	 *                            Default: All broken connections for this service.
 	 */
 	refresh = ( connections = this.props.brokenConnections ) => {
-		this.setState( { isRefreshing: true } );
-
 		this.getConnections( connections ).map( connection => {
-			const keyringConnection = find( this.props.keyringConnections, {
-				ID: connection.keyring_connection_ID,
+			const keyringConnection = find( this.props.keyringConnections, singleKeyringConnection => {
+				// Publicize connections store the keyring ID in keyring_connection_ID
+				// Since keyring connections don't have a keyring_connection_ID property,
+				// we need to check both the keyring_connection_ID and the ID
+				return (
+					( singleKeyringConnection.type === 'publicize' &&
+						singleKeyringConnection.ID === connection.keyring_connection_ID ) ||
+					singleKeyringConnection.ID === connection.ID
+				);
```

Comment:
> I guess there is a very small possibility that the publicize ID of one connection matches the keyring ID of another connection

I don't think this change addresses that issue, since the keyring id being the same as a publicize connection id would still return true.

What about something like
```js
const keyringConnection = find( this.props.keyringConnections, token => {
	// Publicize connections store the token id as `keyring_connection_ID`
	const tokenID =
		'publicize' === token.type ? connection.keyring_connection_ID : connection.ID;
	return token.ID === tokenID;
} );

Reply:
Good point, thanks, I've updated it